tfhe/src/core_crypto/entities/fourier_lwe_bootstrap_key.rs

```diff
@@ -0,0 +1,158 @@
+use crate::core_crypto::prelude::IntoContainerOwned;
+use aligned_vec::{avec, ABox};
+use concrete_fft::c64;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(bound(deserialize = "C: IntoContainerOwned"))]
+pub struct FourierLweBootstrapKey<C: Container<Element = c64>> {
+    fourier: FourierPolynomialList<C>,
+    input_lwe_dimension: LweDimension,
+    glwe_size: GlweSize,
+    decomposition_base_log: DecompositionBaseLog,
+    decomposition_level_count: DecompositionLevelCount,
+}
+
+pub type FourierLweBootstrapKeyView<'a> = FourierLweBootstrapKey<&'a [c64]>;
+pub type FourierLweBootstrapKeyMutView<'a> = FourierLweBootstrapKey<&'a mut [c64]>;
+pub type FourierLweBootstrapKeyOwned = FourierLweBootstrapKey<ABox<[c64]>>;
+
+impl<C: Container<Element = c64>> FourierLweBootstrapKey<C> {
+    pub fn from_container(
+        data: C,
+        input_lwe_dimension: LweDimension,
+        glwe_size: GlweSize,
+        polynomial_size: PolynomialSize,
+        decomposition_base_log: DecompositionBaseLog,
+        decomposition_level_count: DecompositionLevelCount,
+    ) -> Self {
+        assert_eq!(
+            data.container_len(),
+            input_lwe_dimension.0
+                * polynomial_size.to_fourier_polynomial_size().0
+                * decomposition_level_count.0
+                * glwe_size.0
+                * glwe_size.0
+        );
+        Self {
+            fourier: FourierPolynomialList {
+                data,
+                polynomial_size,
+            },
+            input_lwe_dimension,
+            glwe_size,
+            decomposition_base_log,
+            decomposition_level_count,
+        }
+    }
+
+    /// Return an iterator over the GGSW ciphertexts composing the key.
+    pub fn into_ggsw_iter(self) -> impl DoubleEndedIterator<Item = FourierGgswCiphertext<C>>
+    where
+        C: Split,
+    {
+        self.fourier
+            .data
+            .split_into(self.input_lwe_dimension.0)
+            .map(move |slice| {
+                FourierGgswCiphertext::from_container(
+                    slice,
+                    self.glwe_size,
+                    self.fourier.polynomial_size,
+                    self.decomposition_base_log,
+                    self.decomposition_level_count,
+                )
+            })
+    }
+
+    pub fn input_lwe_dimension(&self) -> LweDimension {
+        self.input_lwe_dimension
+    }
+
+    pub fn polynomial_size(&self) -> PolynomialSize {
+        self.fourier.polynomial_size
+    }
+
+    pub fn glwe_size(&self) -> GlweSize {
+        self.glwe_size
+    }
+
+    pub fn decomposition_base_log(&self) -> DecompositionBaseLog {
+        self.decomposition_base_log
+    }
+
+    pub fn decomposition_level_count(&self) -> DecompositionLevelCount {
+        self.decomposition_level_count
+    }
+
+    pub fn output_lwe_dimension(&self) -> LweDimension {
+        LweDimension((self.glwe_size.0 - 1) * self.polynomial_size().0)
```

Comment:
you can now use self.glwe_size.to_glwe_dimension().to_equivalent_lwe_dimension(self.polynomial_size())